### DISCUSSION TITLE
Need help for app.changeCurrentPageBackground()

### DISCUSSION THREAD

**Discussion by Shogun05** [xournalpp/xournalpp#5616] [Q&A] [answered]
I am trying to make a plugin that will default the background of a page to a dark mode page. This is me touching lua for the first time so i am not sure how to go about the `app.changeCurrentPageBackground();` 

The current I have right now in `main.lua` is: 
```lua
function initUi()
    app.registerUi({["menu"] = "Dark Mode", ["callback"] = "darkMode", ["accelerator"] = "<Control><Shift>b"});
  end

  function darkMode()
    app.changeCurrentPageBackground(":image");
  end
  ```

Now i have an image, `bg.png` which is what i wish the background to be. How do I pass arguments into the method to do that?

**Comment by rolandlo** [comment, marked as answer]
It looks to me like the API doesn't support that yet. Basically we need
``` cpp
page->setBackgroundImage(newImg);
page->setBackgroundType(PageType(PageTypeFormat::Image));
```
and we only get the second step via `app.changecurrentPageBackground(":image")`.
